orcid_hub/utils.py

```diff
@@ -650,13 +650,20 @@ def send_user_invitation(inviter,
             disambiguation_source=disambiguation_source,
             token=token)
 
+        status = "The invitation sent at " + datetime.utcnow().isoformat(timespec="seconds")
         if task_type == TaskType.AFFILIATION:
-            status = "The invitation sent at " + datetime.utcnow().isoformat(timespec="seconds")
             (AffiliationRecord.update(status=AffiliationRecord.status + "\n" + status).where(
-                AffiliationRecord.status.is_null(False),
+                AffiliationRecord.status.is_null(False), AffiliationRecord.task_id == task_id,
                 AffiliationRecord.email == email).execute())
             (AffiliationRecord.update(status=status).where(
-                AffiliationRecord.status.is_null(), AffiliationRecord.email == email).execute())
+                AffiliationRecord.task_id == task_id, AffiliationRecord.status.is_null(),
+                AffiliationRecord.email == email).execute())
+        elif task_type in [TaskType.FUNDING, TaskType.WORK, TaskType.PEER_REVIEW]:
+            task = Task.get(task_id)
+            for record in task.records.where(task.record_model.is_active):
+                invitee_class = record.invitees.model
```

Comment:
it's not used anywehre else...

Reply:
"invitee_class"

Reply:
```python
task.record_model.update(status=...').where(
   task.record_model.task == task, task.record_model.is_active,
   task.record_model.emai == emai).execute()
```

Reply:
this will get rid of n+1 a-patter... looping over the records

Reply:
As discussed, invitees cannot be directly called from task object, as they are under each record object of the task, so for now n+1 pattern will be there.

In future, need to figure out how we can enhance the relation between task-->record class(funding/peer-review/work)-->invitees.